File: LaboRTI2023_sources/Data/Article.cpp
#include "Article.h"

Article::Article() : Article(1, "default", 0, "default", 0.0) { }

Article::Article(int id, string intitule, int qte, string img, float prix)
{
    setId(id);
    setIntitule(intitule);
    setQte(qte);
    setImage(img);
    setPrix(prix);
}

Article::Article(const Article& art)
{
    this->setId(art.getId());
    this->setIntitule(art.getIntitule());
    this->setQte(art.getQte());
    this->setImage(art.getImage());
    this->setPrix(art.getPrix());
}


void Article::setId(int id)
{
    if(id < 0)
    {
        throw Exception("Id d'article invalide!");
    }
    this->id = id;
}

void Article::setIntitule(string intitule)
{
    if(intitule.empty())
    {
        throw Exception("Intitule d'article invalide!");
    }
    this->intitule = intitule;
}

void Article::setQte(int qte)
{
    if(qte < 0)
    {
        throw Exception("Quantite d'article invalide!");
    }
    this->quantite = qte;
}

void Article::setImage(string img)
{
    if(img.empty())
    {
        throw Exception("Image d'article invalide!");
    }
    this->image = img;
}

void Article::setPrix(float prix)
{
    if(prix < 0)
    {
        throw Exception("Prix d'article invalide!");
    }
    this->prix = prix;
}



int Article::getId() const { return id; }

string Article::getIntitule() const { return intitule; }

int Article::getQte() const { return quantite; }

string Article::getImage() const { return image; }

float Article::getPrix() const { return prix; }

File: LeMaraicher_ClientAchat/src/Modele/Article.java
package Modele;

public class Article
{
    private int id;
    private String intitule;
    private int quantite;
    private String image;
    private float prix;

    public Article() {
        this(0, "default", 0, "default", 0.0f);
    }

    public Article(int id, String intitule, int quantite, String image, float prix) {
        setId(id);
        setIntitule(intitule);
        setQuantite(quantite);
        setImage(image);
        setPrix(prix);
    }

    public Article(Article art) {
        this(art.getId(), art.getIntitule(), art.getQuantite(), art.getImage(), art.getPrix());
    }

    public void setId(int id) {
        this.id = id;
    }

    public void setIntitule(String intitule) {
        this.intitule = intitule;
    }

    public void setQuantite(int quantite) {
        this.quantite = quantite;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public void setPrix(float prix) {
        this.prix = prix;
    }

    public int getId() {
        return id;
    }

    public String getIntitule() {
        return intitule;
    }

    public int getQuantite() {
        return quantite;
    }

    public String getImage() {
        return image;
    }

    public float getPrix() {
        return prix;
    }
}
